osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -751,13 +750,17 @@ private class CarouselRoot : CarouselGroupEagerSelect
 
             public CarouselRoot(BeatmapCarousel carousel)
             {
+                // root should always remaing selected. if not, PerformSelection will not be called.
+                State.Value = CarouselItemState.Selected;
+                State.ValueChanged += state => State.Value = CarouselItemState.Selected;
+
                 this.carousel = carousel;
             }
 
             protected override void PerformSelection()
             {
                 if (LastSelected == null)
-                    carousel.SelectNextRandom();
+                    carousel?.SelectNextRandom();
```

Comment:
When would ``carousel`` ever be null? ``CarouselRoot`` constructor is never called with null and the value of ``carousel`` is never changed.

Reply:
I guess I should have commented this on peppy's PR instead

Reply:
you can remove the check and run in debug